internal/agent/install.go

```diff
@@ -324,32 +321,26 @@ func ensureDataSourceReady() {
 	}
 }
 
-func prepareConfiguration(ctx context.Context, source string) (string, error) {
+func prepareConfiguration(source string) (io.Reader, error) {
+	var cfg io.Reader
+	// source can be either a file in the system or an url
+	// We need to differentiate between the two
+	// If its a local file, we just read it and return it
+	// If its a url, we need to create a configuration with the url and let the config.Scan handle it
 	// if the source is not an url it is already a configuration path
 	if u, err := url.Parse(source); err != nil || u.Scheme == "" {
-		return source, nil
-	}
-
-	// create a configuration file with the source referenced
-	f, err := os.CreateTemp(os.TempDir(), "kairos-install-*.yaml")
-	if err != nil {
-		return "", err
+		file, err := os.ReadFile(source)
+		if err != nil {
+			return cfg, err
+		}
+		cfg = bytes.NewReader(file)
+		return cfg, nil
 	}
 
-	// defer cleanup until after parent is done
-	go func() {
-		<-ctx.Done()
-		_ = os.RemoveAll(f.Name())
-	}()
-
-	cfg := config.Config{
-		ConfigURL: source,
-	}
-	if err = yaml.NewEncoder(f).Encode(cfg); err != nil {
```

Comment:
may be this was the issue? We never closed the file as the comments in NewEncoder mention:

// NewEncoder returns a new encoder that writes to w.
// The Encoder should be closed after use to flush all data
// to w.


So maybe its was not fully written?

Reply:
cant even find anything else related to this that could explain the issues, unless we have a bug in the collector in which the dir scan gets overwritten fully by the readers scan?

Reply:
will check that just in case as well...